packages/core/src/sync-realtime/service.ts

```diff
@@ -748,21 +723,23 @@ export class RealtimeSyncService extends Emittery<RealtimeSyncEvents> {
    * Check if deep re-org occured by comparing remote "finalized" block to local.
    */
   private reconcileDeepReorg = async (latestBlockNumber: number) => {
```

Comment:
Maybe rename this function "validateFinalizedBlock"? And can prob remove the argument entirely if we get rid of the deepReorg event.